visualize-admin/visualization-tool
# feat: Double-value tooltips (percentage + absolute)

Closes #1093.
Closes #1111.

This PR:
1. adds support for **double-value tooltips** for charts that display percentages (pie, stacked area, stacked column),

**How to test**
- Look for the `Einmalvergütung für Photovoltaikanlagen` dataset.
- Create a stacked column chart.
- Change its mode to `relative figures`.
- Hover over bars to see that the tooltip displays both percentages and original values.
- Change chart type to `area` or `pie` – see that the tooltip still works in this new mode.

2. **improves tooltip positioning** for selected chart types (area, column, line) and fixes the bug with too many event dispatches in the brush component. This was made possible by refactoring the interactive filters state to use Zustand's store instead of React's context to be able to access current state outside of the component context,

**How to test**
- Create an area, column or line chart.
- See that the tooltip is centered when hovering over data points.
- Make the chart stacked (area or line).
- See that the tooltip is attached horizontally to the middle of shape's height and flips the position around 25% width of the chart.

3. **makes the animations _disabled_** during time range filtering (by using another Zustand store for storing the transition duration),

**How to test**
- Look for the `Einmalvergütung für Photovoltaikanlagen` dataset.
- Create a stacked area chart.
- Normalise the chart and see that it's animating.
- Enable interactive time filter from the X field.
- Interact with the interactive filter and see that there is no animation.

4. disables the possibility to enable animations when sorting by total size and vice versa.

**How to test**
- Go to the Bathing water quality dataset.
- Use Parameter as X field.
- Use Bathing site as color (stacked column chart).
- Change the sorting to total size.
- See that the animation is disabled.
- Remove total size sorting.
- Enable animation.
- Go to sorting and see that total size is disabled.

### Notes
See #1113.

_I am going to consolidate the d3 rendering (`join` pattern) and properly disable transitions along the way. Right now it works, but the transition with 0ms duration is still being fired, and a minor lag is noticeable when brushing the area chart on the home page. It should not be a transition with 0ms duration, but rather not be a transition at all._
